src/main/java/org/dcsa/ovs/model/Timestamp.java

```diff
@@ -19,7 +21,8 @@
     private String facilitySMDGCode;
 
     @NotNull
-    private String facilityTypeCode;
+    @EnumSubset(anyOf = {"PBPL", "BRTH"})

```

Comment:
This is designed to work on  a `String`, It will not work over the `FacilityTypeCode` object.

Reply:
With the new DCSA_Core update this should work...